### ISSUE TITLE
New Course Card Layout

### ISSUE THREAD

**Issue by roberthouse54** [mitodl/micromasters#2563]
The goal of this issue is to change the layout in the course card to accomodate a new status area, and a space for course grades.

Invision link: https://invis.io/35AC2AA6J

**Basic Layout**

![image](https://cloud.githubusercontent.com/assets/20047260/23805238/0e81f47a-058b-11e7-8743-b93d826547b7.png)

- The course name is in the upper left. If it is long it will wrap.
- Under the course name are links: 
    - "View on edX"
    - "Contact Course Team" - This is only visible after the user is enrolled, and the feature is behind a paywall (as it is currently).
- Course start date: 
    - If course has not started: "Course starts [date]"
    - If course start date is <10 days: "Course starts in [x] days"
    - If course has started: "In progress"
    - Course ended: "Course has ended"
- Enrollment start date:
    - If enrollment has not started: "Enrollment starts [date]"
    - If enrollment start date is <10 days: "Enrollment starts in [x] days"
    - If enrollment has started: display nothing here.
- If a user is enrolled, display "Enrolled" with a checkmark in the upper right. If the course has ended, do not display this any more.
- Once the user is enrolled, course grades will be displayed as shown in the mock. If there is no grade, display "- -" (as shown.) For programs that do not have a final proctored exam, just display the edx grade. If the course is in progress this will be labelled "edX Progress". If the grade is final, this will be labelled "edX Grade" (for DEDP) and "edX Final Grade (non DEDP).
- When the user has completed the edX grade and the final grade (or in non-DEDP, when the edX Progress becomes the edX Final Grade, we will highlight the fianl grade in a darker color, as shown:
![image](https://cloud.githubusercontent.com/assets/20047260/23810501/ea002896-059f-11e7-98ce-570bdd708491.png)

**Statuses**

![image](https://cloud.githubusercontent.com/assets/20047260/23806026/235b5ed8-058e-11e7-9f4e-55be199dd14d.png)

An area at the bottom of the card will display various status messages. Call-to-Action (CTA) will be a button on the left. Minor CTA will be links. More than one status may be shown at the same time, although this will be rare.

**Auditing**
![image](https://cloud.githubusercontent.com/assets/20047260/23805440/e36f17da-058b-11e7-88c9-3368b4a70db8.png)

Button: "Pay Now"
Copy: "You are auditing. You need to pay to get credit and to take the final exam. (Payment due [date/ within x days].)"
- If payment due date is <10 days, display text in red. (see below)
![image](https://cloud.githubusercontent.com/assets/20047260/23805663/d3ca1270-058c-11e7-8d9a-7300ecbed581.png)

**Passed**
![image](https://cloud.githubusercontent.com/assets/20047260/23805447/ebb8e632-058b-11e7-94a2-5b5795e898d8.png)

Copy: "You passed this course! Click here to view your MicroMasters Certificate"

**Not Passed**
![image](https://cloud.githubusercontent.com/assets/20047260/23805454/f7360346-058b-11e7-87ba-1b788fa248f8.png)

Button: "Re-Enroll"
Copy: 
- "You did not pass this course but you can re-enroll. Next course starts [date/ within x days]. Enrollment open."
- "You did not pass this course but you can re-enroll. Next course starts [date/ within x days]. Enrollment starts [date/ within x days]."
- "You did not pass this course. No future courses are currently scheduled." (no button shown)

**Missed Payment Dedline**
![image](https://cloud.githubusercontent.com/assets/20047260/23805470/fe661cb4-058b-11e7-827c-319149847469.png)

Button: "Re-Enroll"
Copy: "You missed the payment deadline and will not receive credit for the MicroMasters program. You can re-enroll. Next course starts [date/ within x days]."

**edX complete, but need to take exam**
![image](https://cloud.githubusercontent.com/assets/20047260/23805475/04276e14-058c-11e7-8fae-0931ba9dc5be.png)

Button: "Pay Now" (if auditing)
Copy: 
- (If not paid):  "You passed the edX online course. But you still need to pass the final exam to earn MicroMasters credit. Payment is due in 8 days (on Oct 15, 2016). You can take the exam any time between [exam start date] and [exam end date]."
- - (If paid, no button shown):  "You passed the edX online course. But you still need to pass the final exam to earn MicroMasters credit. Payment is complete. You can take the exam any time between [exam start date] and [exam end date]."

**Auditing, but passed a previous course run**
![image](https://cloud.githubusercontent.com/assets/20047260/23805581/775069ea-058c-11e7-80c4-bcf7faa63233.png)

Copy: "You are auditing, but you do not have to pay for this course again because you already completed it. If you want to you can choose to pay for the course [link]."

**Paid but not enrolled error**
![image](https://cloud.githubusercontent.com/assets/20047260/23805484/0f78e3a6-058c-11e7-9945-1a790a841153.png)

Copy: "Something went wrong. You paid for this course but are not currently enrolled. Contact us for help."

**Course ended but payment deadline not passed**
![image](https://cloud.githubusercontent.com/assets/20047260/23805494/18486218-058c-11e7-8f52-fe3a35ae34c3.png)

Button: "Pay Now"
Copy: "This course has ended, but you can still pay for it to receive MicroMasters credit."

**Coupon**
![image](https://cloud.githubusercontent.com/assets/20047260/23805926/cf38ecc6-058d-11e7-972f-c032ee00d6f2.png)

Copy: "You have a coupon for [discount] off the normal price of this course."

**Past Course Runs**
If a user is enrolled in a course that he/she has taken in the past, we will display "Past completed course runs" at the bottom of the card. This section will be closed by default. Clicking the header or arrow opens the section. The layout is mostly the same as we have currently, but with out the gray background color.

![image](https://cloud.githubusercontent.com/assets/20047260/23809599/462a90c4-059c-11e7-9567-30eee64cca8e.png)

**Comment by pdpinch** [comment]
This is blocked by DB's PR

**Comment by pdpinch** [comment]
I moved this back to `In Design` because Rob's going to update this with new images.

**Comment by roberthouse54** [comment]
I am closing this... I will be re-writing it as a new issue.
